File: CuriosityApp/app/src/main/java/ziemansoft/ziemapp/curiosityapp/MainActivity.java
package ziemansoft.ziemapp.curiosityapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import org.json.JSONObject;

import ziemansoft.ziemapp.curiosityapp.networkUtils.LinkData;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        JSONObject jsonObject = LinkData.getJsonObjectFromNetwork();
        if (jsonObject == null) {
            Log.d("MyResult", "not working");
        } else {
            Log.d("MyResult", "working");

        }
    }
}

File: CoinsOnlineMVP/app/src/main/java/ziemansoft/ziemapp/coins_onlinemvp/MainActivity.kt
package ziemansoft.ziemapp.coins_onlinemvp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import ziemansoft.ziemapp.coins_onlinemvp.adapters.CoinsAdapter
import ziemansoft.ziemapp.coins_onlinemvp.models.MainMetheds
import ziemansoft.ziemapp.coins_onlinemvp.presenters.MainPresenter
import ziemansoft.ziemapp.coinsonline.pojo.CoinPriceInfo

class MainActivity : AppCompatActivity(), MainMetheds {
    val mainPresenter = MainPresenter(this)
    val adapter = CoinsAdapter(this)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        recyclerView.adapter = adapter
        mainPresenter.loadData()
    }

    override fun loadData(coinsInfo: List<CoinPriceInfo>) {
        adapter.coinInfoList = coinsInfo
    }
}